mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java

```diff
@@ -342,6 +342,13 @@ public void onBackPressed() {
         if (mDrawerLayout != null && mDrawerLayout.isDrawerOpen(GravityCompat.START)) {
             mDrawerLayout.closeDrawer(GravityCompat.START);
         } else {
+            String currentFragment = getSupportFragmentManager()
+                    .findFragmentById(R.id.container)
+                    .getClass().getSimpleName();
+            if (!currentFragment.equalsIgnoreCase("SearchFragment")) {
```

Comment:
Ignore hardcoded strings, if necessary declare it in constants util

Reply:
Should I change it in switch block too on line `266` same file.

Reply:
@robustTechie I have made the necessary changes. Please take a look.

Reply:
please mark this as resolved @iamsh4shank , Thanks